### PR TITLE
[Merged by Bors] - chore: sync 'ite' and 'dite' with lean4

### PR DESCRIPTION
**Pull request by dselsam**
Corresponding Mathlib commit is https://github.com/dselsam/mathlib/commit/fe97d81b6d0f89b229c7dfea36554d94f5e2dfc1

### CODE REVIEW

**Comment by bryangingechen** [comment]
In lieu of running `bors try` here, here's a link to the results of CI on the first commit, which shows 6 broken tests: https://github.com/dselsam/lean3/runs/1884575459#step:9:2914

**Comment by dselsam** [comment]
Sorry, rusty on the protocol. Fixed.

File: tests/lean/eformat.lean [hunk 1]
```diff
@@ -11,32 +11,32 @@ meta def trace_tagged_debug_locals : tactic unit := do
   cs ← cs.mmap infer_type,
   cs.mmap' trace_tagged_debug
 
-example (α : Type) (x : α) (h : 1 = 2) (f : ite ff true false) (g : x ≠ x) : true :=
+def example1 (α : Type) (x : α) (h : 1 = 2) (f : ite ff true false) (g : x ≠ x) : true :=
```
Comment (bryangingechen) [inline tests/lean/eformat.lean:14]: Is there a `ite`/`dite`-related reason why these were changed to `def`s?

Comment (dselsam) [reply tests/lean/eformat.lean:14]: It is an unrelated fix. The test compares against an expected output, but the order of the traces was nondeterministic as written.

Comment (gebner) [reply tests/lean/eformat.lean:14]: The *order* should always be deterministic (the tests run with `-j0`).  Are you maybe referring to some automatically generated names?

Comment (dselsam) [reply tests/lean/eformat.lean:14]: I was typing a clarification :) I was debugging the test manually (it broke for `ite`-related reasons), and the nondeterminism was annoying. I realized after writing my previous comment that probably many tests rely on each test being run with a single thread, in which case I should probably revert this change.

Comment (dselsam) [reply tests/lean/eformat.lean]: Pushed a revert.

**Review by bryangingechen — APPROVED** [review]
I'm happy with this modulo the C++ changes that I didn't check.

**Pull request closed without merging**